packages/flutter/test/flutter_test_config.dart

```diff
@@ -19,6 +19,18 @@ Future<void> testExecutable(FutureOr<void> Function() testMain) {
   // receive the event.
   WidgetController.hitTestWarningShouldBeFatal = true;
 
+  // Custom matchesGoldenFile wrapper for handling flaky tests
+  Future<void> expectFlutterGoldenMatches(
+    Object key,
+    String goldenFile, {
+    bool isFlaky = false,
+  }) {
+    if (isFlaky) {
+      (goldenFileComparator as flutter_goldens.FlutterGoldenFileComparator).testIsFlaky();
```

Comment:
Also @yjbanov instead of setting a boolean flag here, you could pass the `goldenFile` and the comparator could handle the flaky test that way.